WebKit's "Copy extension" build phase does not run in engineering builds

aestes opened the pull request:
#### 90db5bec95cdb7a1069b4078110884f9c6a6be3a
<pre>
WebKit&apos;s &quot;Copy extension&quot; build phase does not run in engineering builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=267318">https://bugs.webkit.org/show_bug.cgi?id=267318</a>
<a href="https://rdar.apple.com/120771873">rdar://120771873</a>

Reviewed by Per Arne Vollan.

WebKit&apos;s &quot;Copy extension (temporary workaround)&quot; script build phase is designed for Apple&apos;s
production build system, but exits early when run from an Xcode build. As a result, the app
extensions in $BUILT_PRODUCTS_DIR do not have the expected modifications to their Info.plists.

Resolved this by teaching the script to account for Xcode builds by changing EXTENSIONS_PATH to be
$BUILT_PRODUCTS_DIR instead of the extensions&apos; $INSTALL_PATH and skipping the copying of extensions
to $DSTROOT.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/272854@main">https://commits.webkit.org/272854@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cda93de24f00ab49dda63558283970625209f6c2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33346 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12117 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35255 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35972 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30311 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14462 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9261 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29445 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33820 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10211 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29765 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8908 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9036 "Found 2 new test failures: imported/w3c/web-platform-tests/beacon/beacon-cors.https.window.html, imported/w3c/web-platform-tests/mixed-content/gen/worker-classic-data.http-rp/opt-in/fetch.https.html (failure)") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37302 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30275 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30096 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35156 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9159 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7105 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33021 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10907 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7724 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9741 "Built successfully") | | | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/22/builds/4280 "The change is no longer eligible for processing.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9882 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 55;
```
Should we revert these unrelated changes?

aestes replied:
I actually think it's OK to land these changes. Otherwise, Xcode will keep making them whenever you modify this project.

emw-apple replied:
Looked it up, and I think this corresponds to Xcode 13.0, so yes, it should be fine.

pvollan reviewed (approved):
R=me.

pvollan commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18913,7 +18913,9 @@
 		5C400E6A29DB8AB500446F6F /* PBXTargetDependency */ = {
 			isa = PBXTargetDependency;
-			platformFilter = ios;
+			platformFilters = (
+				ios,
+			);
```
Ditto.

aestes replied:
I actually think it's OK to land these changes. Otherwise, Xcode will keep making them whenever you modify this project.

pvollan replied:
Does older Xcode versions support this, or do we not need to maintain compatibility?

aestes replied:
Yes they support this.

emw-apple reviewed (approved):
rs=me, we've discussed that this is a temporary build fix. @pvollan is going to change the base values in Info.plist since we are now patching them in all configurations.

aestes committed:
WebKit's "Copy extension" build phase does not run in engineering builds
https://bugs.webkit.org/show_bug.cgi?id=267318
rdar://120771873

Reviewed by Per Arne Vollan.

WebKit's "Copy extension (temporary workaround)" script build phase is designed for Apple's
production build system, but exits early when run from an Xcode build. As a result, the app
extensions in $BUILT_PRODUCTS_DIR do not have the expected modifications to their Info.plists.

Resolved this by teaching the script to account for Xcode builds by changing EXTENSIONS_PATH to be
$BUILT_PRODUCTS_DIR instead of the extensions' $INSTALL_PATH and skipping the copying of extensions
to $DSTROOT.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/272854@main
